lib/shopping_cart.rb

```diff
@@ -0,0 +1,46 @@
+require './lib/product'
+
+class ShoppingCart
+  attr_reader :name,
+              :capacity,
+              :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
```

Comment:
This works and is good, but you can just go ahead and create the hash with the keys and values you want by doing 
` {name: @name, capacity: @capacity}`